src/riscv/lib/src/pvm/linux.rs

```diff
@@ -271,7 +272,12 @@ where
             .map(|(addr, data)| addr.saturating_add(data.len() as u64))
             .max()
             .unwrap_or(0);
-        let program_length = program_end.saturating_sub(program_start) as usize;
+
+        let Some(program_length) = NonZeroU64::new(program_end.saturating_sub(program_start))
+        else {
+            // there is no program to load
+            return Ok(());
```

Comment:
There is a bit of state that doesn't get set if you exit here. You could either move that up or use conditionals on the blocks that require `program_length`.

Reply:
done [diff](https://github.com/tezos/riscv-pvm/compare/7c0394853287802ad5c045c8c35c6d52b1867f1e..cb79f02099b4bf62a908003bf3535c78d0e2d8e4)